src/main/java/com/poliscrypts/api/security/UserDetailsServiceImpl.java

```diff
@@ -0,0 +1,33 @@
+package com.poliscrypts.api.security;
+
+import com.poliscrypts.api.entity.User;
+import com.poliscrypts.api.repository.UserRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.authority.SimpleGrantedAuthority;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.stereotype.Service;
+
+import javax.transaction.Transactional;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Service
+@Transactional
```

Comment:
i didn't see rollbacks so why you use it?

Reply:
ditto